scripts/create-instance.sh

```diff
@@ -0,0 +1,391 @@
+cf create-app $webapp_name
+##
+
+##
+# Create Services
+##
+cf create-service aws-rds micro-psql "${service_name_rds}"
+cf create-service aws-elasticache-redis redis-dev "${service_name_redis}"
+cf create-service s3 basic-public "${service_name_s3}"
+cf create-service s3 basic "${service_name_s3_build_logs}"
+cf create-service cloud-gov-service-account space-deployer "${service_name_cf_api_user}"
+cf create-service-key "${service_name_cf_api_user}" "${service_name_cf_api_user}-key"
+cf create-service cloud-gov-service-account space-deployer "${service_name_cf_ci_user}"
+cf create-service-key "${service_name_cf_ci_user}" "${service_name_cf_ci_user}-key"
+cf create-user-provided-service "${service_name_sitewide_error}" -p '{"BODY": "", "HEADING": ""}'
+cf create-user-provided-service "${service_name_sqs}" -p '{"sqs_url": ""}'
+cf create-user-provided-service "${service_name_uev_key}" \
+  -p "$(cat <<- EOF
+    {
+      "key": "$(openssl rand -hex 20)"
+    }
+EOF
+)"
+
+cf create-user-provided-service "${service_name_env}" \
+  -p "$(cat <<- EOF
+    {
+      "FEDERALIST_SESSION_SECRET": "$(openssl rand -hex 25)",
+      "GITHUB_CLIENT_CALLBACK_URL": "https://${domain}/auth/github/callback",
+      "GITHUB_CLIENT_EXTERNAL_CALLBACK_URL": "https://${domain}/external/auth/github/callback",
+      "GITHUB_CLIENT_ID": "${GITHUB_CLIENT_ID}",
+      "GITHUB_CLIENT_SECRET": "${GITHUB_CLIENT_SECRET}",
+      "GITHUB_WEBHOOK_SECRET": "$(openssl rand -hex 20)",
+      "GITHUB_WEBHOOK_URL": "https://${domain}/webhook/github"
+    }
+EOF
+)"
+
+cf create-user-provided-service "$service_name_proxy" \
+  -p "$(cat <<- EOF
+    {
+      "guid": "$(cf app ${proxy_name} --guid)"
+    }
+EOF
+)"
+
+cf create-user-provided-service "${service_name_domain}" \
+  -p "$(cat <<- EOF
+    {
+      "guid": "$(cf curl "/v3/domains?names=${proxy_domain}" | jq -r '.resources[0].guid')"
+    }
+EOF
+)"
+
+cf create-user-provided-service "${service_name_space}" \
+  -p "$(cat <<- EOF
+    {
+      "guid": "$(cf space ${cf_space} --guid)"
+    }
+EOF
+)"
+
+api_user_password=$(cf service-key "${service_name_cf_api_user}" "${service_name_cf_api_user}-key" | tail -n +2 | jq -r '.password')
+api_user_username=$(cf service-key "${service_name_cf_api_user}" "${service_name_cf_api_user}-key" | tail -n +2 | jq -r '.username')
+cf create-user-provided-service "${service_name_cf_creds}" \
+  -p "$(cat <<- EOF
+    {
+      "DEPLOY_USER_PASSWORD": "${api_user_password}",
+      "DEPLOY_USER_USERNAME": "${api_user_username}"
+    }
+EOF
+)"
+
+# Grant permissions in the sites space
+cf set-space-role "${api_user_username}" "${cf_org}" "${cf_sites_space}" "SpaceDeveloper"
+
+# Temporarily create a service key to get the bucket name
+tmp_s3_service_key_name="${service_name_s3}-key-tmp"
+cf create-service-key "${service_name_s3}" "${tmp_s3_service_key_name}"
+shared_bucket=$(cf service-key "${service_name_s3}" "${tmp_s3_service_key_name}" | tail -n +2 | jq -r .bucket)
```

Comment:
Not sure we need a shared bucket in new space creation.

Reply:
Just legacy to deploy the proxy?

Reply:
yeah, we still have a few stranglers, we can remove the shared bucket as a separate issue.